_posts/2022-04-18-Diffie-Hellman.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: Diffie Hellman
+author: Charles Thomas
+---
+
+As I've been studying Quantum Computing I came across Diffie Hellman but it took me a while to get my head around it. So I thought I would try to explain it.
+
+# Symmetric Ciphers
+To understand Diffie Hellman let's first talk about the concept of the symmetric cipher. A symmetric cipher is when we encrypt a message with one key and then use the same key to decrypt it.
+
+As an example let's look at the Caesar cipher. The cipher starts by giving each letter a number. For example a becomes 1, b becomes 2 and so on. This means we can take a message made up of numbers and write it in letters. For example, hello becomes 8 5 12 12 15
+
+Then to encrypt the message we add n to all the numbers. If n is 3 then our message would become 11 8 15 15 18. The important caveat is that if our number is bigger than 26 we go back to counting at 1. So if we have the letter y it has the number 25, then if we had 3 to it we would normally get 28 but in our case we get 2. (This is called modular arithmetic and we'll talk about more about it later).
+
+So we now have our encrypted numbers 11 8 15 15 18. Let's turn it back into letters to get khoor. We can then send khoor to anyone we want. But if they want to decrypt it they need to know n i.e. how many places which shifted the letters.
+
+So we need a way of safely sending n to a person so we can communicate using our ceaser cipher. Diffie Hellman allows us to solve this problem.
```

Comment:
In order to communicate with another person, we need to tell them which number n to use.